src/NuGet.Core/NuGet.Protocol/Plugins/InboundRequestContext.cs

```diff
@@ -37,7 +38,7 @@ public sealed class InboundRequestContext : IDisposable
             IConnection connection,
             string requestId,
             CancellationToken cancellationToken)
-            : this(connection, requestId, cancellationToken, PluginLogger.DefaultInstance)
+            : this(connection, requestId, cancellationToken, new InboundRequestProcessingHandler(), PluginLogger.DefaultInstance)
```

Comment:
All these types are internal. 
So no one but NuGet can use a dedicated handler.
I did that on purpose because others do not need to make that decision, only the client does.